pages/querying/best-practices.mdx

```diff
@@ -0,0 +1,778 @@
+{<h4>Return only a property</h4>} 
+
+Here is a query that extracts nodes' internal ids from the path:
+```cypher
+MATCH path=(:Character { name: "Jon Snow" })-[:KILLED *bfs]->(:Character)
+WITH extract(n IN nodes(path) | ID(n)) AS nodes_ids
+RETURN nodes_ids;
+```
+
+When run in Memgraph Lab, **the full roundtrip is down to 41 ms**. 
+
+![](/pages/querying/best-practices/query-optimization-lab-got-4.png)
+
+Similarly, the following query returns only `name` nodes' property from the path:
+```cypher
+MATCH path=(:Character { name: "Jon Snow" })-[:KILLED *bfs]->(:Character)
+WITH extract(n IN nodes(path) | n {.name}) AS nodes_ids
+RETURN nodes_ids;
+```
+
+When run in Memgraph Lab, **the full roundtrip is down to 31 ms**. 
+
+![](/pages/querying/best-practices/query-optimization-lab-got-5.png)
+
+These are some of the tips that can help to reduce the roundtrip time. **On larger
+datasets, these optimization strategies have larger impact and should be
+utilized whenever possible to achieve the best performance**. 
+
+## Optimizing deep-path traversals
+In contrast to other graph databases, Memgraph deep-path traversals efficiently
+handle complex graph queries, as these algorithms have been built into
+Memgraph's core. This eliminates the need for the overhead of business logic on
+the application side. There are four built-in deep path traversal algorithms:
+Depth-first search (DFS), Breadth-first search (BFS), Weighted Shortest Path and
+All Shortest Paths. 
+
+<Cards>
+    <Card
+    title="Read more about built-in algorithms"
+    href="/advanced-algorithms/built-in-graph-algorithms"
+    />
+</Cards>
+
+Deep path traversal algorithms play a crucial role in exploring and analyzing 
+graph data. In Memgraph, a powerful graph database management system, optimizing 
+these algorithms can significantly enhance performance when dealing with 
+large-scale graph structures. 
+
+For example, let's take the following unoptimized query that traverses through the 
+European transportation network. Using the BFS algorithm, we'll find all of the cities 
+London is connected to with the shortest road path available. In other words, we'll 
+find all shortest paths from the starting node with the `name` property being `London` 
+to all of the nodes in the dataset. 
+
+```cypher
+MATCH path=(:City {name: 'London'})-[*BFS]->(:City)
+RETURN path;
+```
+
+This query returns nodes representing cities London is connected to with the 
+transportation network. 
+
+Since we didn't provide any restrictions and filtering, the algorithm scans and 
+traverses through entire dataset which can lead to slower performance on a larger 
+scale datasets.
+
+<Callout> 
+**Limit query memory usage** 
+
+It is a good practice to limit [query memory
+usage](/fundamentals/storage-memory-usage#control-query-memory-usage) before
+running deep path traversals, to avoid large memory consumption for unoptimized
+queries. Keep in mind that query execution memory monotonically grows in size
```

Comment:
monotonically is strange word for me, but maybe it is just me.

Reply:
I agree, I just c/p that, I will update it. Thanks